### PR TITLE
Update to 2022.1.0

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- Update version following PR 6
- Add compiler test, taken from cf
- Change selector to use linux instead of linux64 (as there are issues with linux64)
- Added activation scripts for linux.

Tests:
- Working on updating feedstock dpctl using this compiler. There are still issues there but at the moment it seems to be on dpctl side.

PS: the linter trips on this recipe.

### CODE REVIEW

**Comment by cbouss** [comment]
linux-64 fails with a crash when running compiled simple program, likely when calling sycl code.
```
         INFO   SYCL_LANGUAGE_VERSION = 202001                                                                                                                                                                                      
0:19:32  ERROR  Command failed with exit code 1      
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -335,15 +363,15 @@ outputs:
...
         - dir %PREFIX%\Library\lib\*    # [win64]
 
   - name: mkl-devel-dpcpp
-    script: repack.sh   # [linux64]
+    script: repack.sh   # [linux]
     script: repack.bat  # [win64]
     build:
-      number: {{ mkl_dpcpp_build_number|int + dst_build_number|int }}   # [linux64 or win64]
-      skip: True  # [not (linux64 or win64)]
+      number: {{ mkl_dpcpp_build_number|int + dst_build_number|int }}   # [linux or win64]
```
Comment (katietz) [inline recipe/meta.yaml:373]: most changes here are in that broken and/or incomplete transition of selectors ... not sure if we need to spend time on such changes.  Especially if they make things not the better

**Review by katietz — CHANGES_REQUESTED** [review]

File: abs.yaml [hunk 2]
```diff
@@ -3,3 +3,6 @@ build_parameters:
   - "--suppress-variables"
   - "--error-overlinking"
   - "--error-overdepending"
+
+channels: 
+  cbouss: dpcpp
```
[the remark this answers is not in the document]
Comment (cbouss) [reply abs.yaml]: removed

File: recipe/conda_build_config.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+CHOST:                                              # [linux and x86]
+  - x86_64-conda-linux-gnu                          # [linux and x86]
+FINAL_PYTHON_SYSCONFIGDATA_NAME:                    # [linux and x86]
+  - _sysconfigdata_x86_64_conda_linux_gnu           # [linux and x86]
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/conda_build_config.yaml]: Changed all selectors to win, osx and linux. 
Updated main build skip to # [not x86].

File: recipe/meta.yaml [hunk 4]
```diff
@@ -37,17 +37,19 @@ source:
   - url: https://anaconda.org/intel/intel-opencl-rt/{{ version }}/download/{{ target_platform }}/intel-opencl-rt-{{ version }}-intel_{{ intel_build_number }}.tar.bz2
     folder: intel-opencl-rt
   {% endif %}
-  {% if linux64 or win64 %}
+  {% if linux or win64 %}
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml]: Changed all selectors to win, osx and linux. 
Updated main build skip to # [not x86].

File: recipe/meta.yaml [hunk 5]
```diff
@@ -37,17 +37,19 @@ source:
...
     folder: dpcpp_{{ target_platform }}
+    patches:
+      - patches/0001-dpcpp-activation.patch # [linux]
   - url: https://anaconda.org/intel/mkl-dpcpp/{{ version }}/download/{{ target_platform }}/mkl-dpcpp-{{ version }}-intel_{{ mkl_dpcpp_build_number }}.tar.bz2
     folder: mkl-dpcpp
   - url: https://anaconda.org/intel/mkl-devel-dpcpp/{{ version }}/download/{{ target_platform }}/mkl-devel-dpcpp-{{ version }}-intel_{{ mkl_dpcpp_build_number }}.tar.bz2
     folder: mkl-devel-dpcpp
   {% endif %}
-  {% if linux64 %}
+  {% if linux %}
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml:50]: Changed all selectors to win, osx and linux. 
Updated main build skip to # [not x86].

File: recipe/meta.yaml [hunk 6]
```diff
@@ -206,17 +213,25 @@ outputs:
     script: repack.sh   # [unix]
     script: repack.bat  # [win]
     build:
-      skip: True  # [not (linux64 or win64)]
+      skip: True  # [not (linux or win64)]
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml]: Changed all selectors to win, osx and linux. 
Updated main build skip to # [not x86].

File: recipe/meta.yaml [hunk 7]
```diff
@@ -239,49 +254,62 @@ outputs:
...
 
-  - name: dpcpp_{{ target_platform }}
-    script: repack.sh   # [unix]
-    script: repack.bat  # [win]
+  - name: dpcpp_{{ target_platform }}    
+    script: update-activate-and-repack.sh   # [unix]
+    script: repack.bat                      # [win]
     build:
-      skip: True  # [not (linux64 or win64)]
+      skip: True  # [not (linux or win64)]
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml]: Changed all selectors to win, osx and linux. 
Updated main build skip to # [not x86].

File: recipe/meta.yaml [hunk 8]
```diff
@@ -239,49 +254,62 @@ outputs:
...
+          - {{ pin_subpackage("dpcpp-cpp-rt", max_pin="x") }}          
+          - libgcc-ng >={{ c_compiler_version }}      # [linux]
+          - libstdcxx-ng >={{ cxx_compiler_version }} # [linux]
+    requirements:   # [linux or win64]
+      # for us to be able to use clang/icx -dumpmachine
+      build:        # [linux or win64]
+        - dpcpp_impl_{{ target_platform }}  # [linux or win64]
+      run:          # [linux or win64]
+        - {{ pin_subpackage('dpcpp_impl_linux-64', exact=True) }} # [linux]
         - {{ pin_subpackage('dpcpp_impl_win-64', exact=True) }}   # [win64]
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml]: Changed all selectors to win, osx and linux. 
Updated main build skip to # [not x86].

File: recipe/meta.yaml [hunk 9]
```diff
@@ -302,11 +330,11 @@ outputs:
         - ls -A1 $PREFIX/lib/*  # [unix]
 
   - name: mkl-dpcpp
-    script: repack.sh   # [linux64]
+    script: repack.sh   # [linux]
     script: repack.bat  # [win64]
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml]: Changed all selectors to win, osx and linux. 
Updated main build skip to # [not x86].

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: Changed all selectors to win, osx and linux. 
Updated main build skip to # [not x86].

**Review by katietz — APPROVED** [review]
thanks

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**